artifactory/commands/buildinfo/addgit_test.go

```diff
@@ -62,32 +64,75 @@ func TestBuildAddGitSubmodules(t *testing.T) {
 		t.Run(test, func(t *testing.T) {
 			tmpBuildName := test + "-Build-" + strconv.FormatInt(time.Now().Unix(), 10)
 			err := runBuildAddGit(t, tmpBuildName, "1", projectPath, test == "dotGitProvided")
+			require.NoError(t, err)
+			partials := getBuildInfoPartials(t, tmpBuildName, "1", "")
+			assertVcsSubmodules(t, partials)
+		})
+	}
+}
+
+func TestBuildAddGitVCSDetails(t *testing.T) {
+	bagTests := []struct {
+		name        string
+		originalDir string
+		revision    string
+		branch      string
+		message     string
+	}{
+		{"Test vcs details without branch", withGit, "6198a6294722fdc75a570aac505784d2ec0d1818", "", "TEST-2 - Adding text to file1.txt"},
+		{"Test vcs details with branch", withBranch, "b033a0e508bdb52eee25654c9e12db33ff01b8ff", "master", "TEST-4 - Adding text to file2.txt"}}
+
+	for _, test := range bagTests {
+		t.Run(test.name, func(t *testing.T) {
+			baseDir, dotGitPath := tests.PrepareDotGitDir(t, test.originalDir, filepath.Join("..", "testdata"))
+			buildDir := getBuildDir(t)
+			checkFailureAndClean(t, buildDir, dotGitPath, test.originalDir)
```

Comment:
See below commend for explanation
```suggestion
			defer cleanUp(t, buildDir, dotGitPath, test.originalDir)
```